### PR TITLE
Transaction.decode/1

### PR DESCRIPTION
**Pull request by robrobbins**
Just getting started here. Will spend a few hours each working day if possible getting it done. Shouldn't take too long as we already have a working case in my day job at Coinlist (for legacy, will be adding 1559 next week) 

Will remain in draft as i like to psuedocode and comment to discuss before going down a path too far

### CODE REVIEW

**Comment by robrobbins** [comment]
haven't forgotten this. just busy... i'll get there

File: lib/ethers/transaction.ex [hunk 1]
```diff
@@ -109,6 +109,50 @@ defmodule Ethers.Transaction do
...
+      Enum.take(vector, 9)
+      |> ExRLP.encode()
+      |> prepend_type_envelope(:eip1559)
+      |> ExKeccak.hash_256()
+
+    [y, r, s] = Enum.take(vector, -3)
+
+    {:ok, public_key} = ExSecp256k1.recover(tx_hash, r, s, :binary.decode_unsigned(y))
+
+    {:ok, struct!(__MODULE__, %{
```
Comment (robrobbins) [inline lib/ethers/transaction.ex:134]: @alisinabh i'm assuming we want to return a transaction struct from `decode/1`

Comment (alisinabh) [reply lib/ethers/transaction.ex:134]: Hey, Sorry I somehow missed this.

Yes it should.

Do you still want to work on this?

**Comment by alisinabh** [comment]
Closing in favour of #166

**Pull request closed without merging**